### ISSUE TITLE
How to quantize the format of h5 by API tf.keras ?

### ISSUE THREAD

**Issue by dlml** [tensorflow/tensorflow#22881]
import numpy as np
import tensorflow as tf

**the code only generate a float tflite, but i want to get an int8 tflite, i dont't kown how to do** 
if somebody can solve this issue [https://github.com/tensorflow/tensorflow/issues/22880](url) , maybe this problem can be solved.

# Generate tf.keras model.
model = tf.keras.models.Sequential()
model.add(tf.keras.layers.Dense(2, input_shape=(3,)))
model.add(tf.keras.layers.RepeatVector(3))
model.add(tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(3)))
model.compile(loss=tf.keras.losses.MSE,
              optimizer=tf.keras.optimizers.RMSprop(lr=0.0001),
              metrics=[tf.keras.metrics.categorical_accuracy],
              sample_weight_mode='temporal')

x = np.random.random((1, 3))
y = np.random.random((1, 3, 3))
model.train_on_batch(x, y)
model.predict(x)

# Save tf.keras model in HDF5 format.
keras_file = "keras_model.h5"
tf.keras.models.save_model(model, keras_file)

# Convert to TensorFlow Lite model.
converter = tf.contrib.lite.TFLiteConverter.from_keras_model_file(keras_file)
tflite_model = converter.convert()
open("converted_model.tflite", "wb").write(tflite_model)

**Comment by dlml** [comment]
@wt-huang

**Comment by wt-huang** [comment]
@dlml: @aselle correct me if I am off, but I don't think keras API supports that at the moment. To get int8 model you can first load the above `keras_model.h5` file using [tftables](https://github.com/ghcollin/tftables) to create TensorFlow graph. Next freeze the graph into .pb file. Then you can convert the resulting graph to int8 tflite using toco.

**Comment by ZJPUBG** [comment]
@wt-huang correct me if I'm wrong. I thought for toco to output quantized tflite, you have to give it an already quantized .pb file? Just giving a .pb file would get a non-quantized tflite file.

**Comment by wt-huang** [comment]
@ZJPUBG In the last step I mentioned, invoke toco to convert .pb into int8 tflite FlatBuffer, make sure to set
`--inference_type=QUANTIZED_UINT8` and `--inference_input_type=QUANTIZED_UINT8`

**Comment by ZJPUBG** [comment]
@wt-huang Thanks for your prompt reply. I know this last step. But my current understanding is, in order to set these flags you mentioned, the training process has to be a quantization-aware training, right? I guess this is the most confused part for me. Thanks.

**Comment by wt-huang** [comment]
@ZJPUBG Yes, quantization-aware training would be ideal in the first step to quantize weights and activation tensors, otherwise final classification results may not be correct. Alternatively, you can use post-training quantization, which is more widely available but could potentially suffer from accuracy loss.
